### ISSUE TITLE
Import error messages & add Types? 

### ISSUE THREAD

**Issue by weilinzung** [chrisblossom/read-dir-deep#1]
Hi,

Try to do `import readDirDeep from 'read-dir-deep'`, but get some error messages:

```
Type 'string' is not assignable to type 'string[]'.ts(2322)
read-dir-deep.d.ts(2, 5): The expected type comes from property 'patterns' which is declared here on type 'Options'
```

also would be great to add types. thanks!

**Comment by chrisblossom** [comment]
`read-dir-deep` is written in TS and exports types as part of the build process.

Could you possibly provide a minimal reproducible example of how you are getting this error? I am unable to reproduce.

**Comment by weilinzung** [comment]
@chrisblossom 

Here is the example I am using, basically all the `options` would not works and gave me an error like this. However, if I just use `const readDirDeep = require('read-dir-deep');` which would be fine:

```
(property) ignore: string[]
Argument of type '{ ignore: string[]; }' is not assignable to parameter of type 'Options'.
  Object literal may only specify known properties, and 'ignore' does not exist in type 'Options'.ts(2345)
```
---------------------------

```
import readDirDeep from 'read-dir-deep';

files =  await readDirDeep('dir', {
 absolute: true,
 patterns: '**/*.{html,css,js}',
 ignore: ['index.html']
};
console.log(files);
```

**Comment by chrisblossom** [comment]
`4.0.4` has been released to npm [fixing this issue](https://github.com/chrisblossom/read-dir-deep/commit/eb79ee76ba9dbc08bc403dc3f591ca7649b5fa60). Also, `patterns` should be an array.

FYI, I've also released [`5.0.0`](https://github.com/chrisblossom/read-dir-deep/commit/3c91e711963c804b7b53bd91597355c62d34214a) which removes the default export and adds the named exports: `readDirDeep` and `readDirDeepSync`. `readDirDeep.sync` has been removed.

Thanks for the report! Please verify your issue is fixed.

**Comment by weilinzung** [comment]
@chrisblossom awesome. That fixed and better. Thanks!
